Console/Command/QueueShell.php

```diff
@@ -250,6 +255,126 @@ public function stats() {
+	 *
+	 * @return null
+	 */
+	public function statusFile() {
+		$info = $this->getStatus();
+		if ($info['status'] == 'bad') {
+			$this->sendAlert($info);
+		}
+		$this->writeFile($info);	
+	}
+
+	/**
+	 * Fetch the status from the queue database and put data in an array the different output functions can use.
+	 *
+	 * @return array Keyed array of metrics we want from the queue.
+	 */
+	private function getStatus() {
+		$info['queue_size'] = $this->QueuedTask->getPending();
+		$lastTask = $this->QueuedTask->getLastCompleted();
+		if (!empty($lastTask)) {
+			$info['last_task_completed'] = $lastTask['QueuedTask']['completed'];
+		} else {
+			$info['last_task_completed'] = null;
+		}
+		$info['status'] = 'good';
+		if ($info['queue_size'] > self::ALERT_SIZE) {
+			$info['status'] = 'bad';
+		}
+		return $info;
+	}
+
+	/**
+	 * Write the output of status() to a selected file path.
+	 * Where the file is written is defined in app/Config/bootstrap.{environment}.php
+	 *
+	 * @param array $info The values to output
+	 */
+	private function writeFile($info) {
+		App::import('File');
+		$file = new File(QUEUE_MONITOR_OUTFILE);
+		if (!$file->write('readJson(' . json_encode($info) . ")\n", $mode = 'w', $force = true)) {
+			$this->out('Unable to write JSONP file.');
+			CakeLog::error('CakephpQueue Monitor: Unable to write JSONP file.');
+		}
+		$file->close();
+		$this->out('File written to: ' . QUEUE_MONITOR_OUTFILE);
+	}
+	
+	/**
+	 * Send Alert.  This could be an email or message to HipChat.
+	 * For this version it goes to HipChat
+	 * 
+	 * @param array $info Array of queue info.
+	 */
+	private function sendAlert($info) {
+		if ($this->spamCheck()) {
+			$this->sendToHipchat(ENVIRONMENT . ' -- Queue has ' . $info['queue_size'] . ' pending items.');
+			$this->out('Alert triggered.');
+		}
+	}
+
+	/**
+	 * Make sure we aren't sending the same message over and over.
+	 *
+	 * @return boolean true if we are not spamming, false if we do not want to send message.
+	 */
+	private function spamCheck() {
+		App::import('File');
+		$file = new File('tmp/queue_last_alert.txt');
+		if ($file->exists()) {
+			$last_alert = strtotime($file->read());
+			if ((time() - $last_alert) < self::ALERT_COOLDOWN) {
+				$this->out('Alert triggered, but on cooldown.');
+				return false;
+			}
+		}
+		if ($file->write(date('c'), $mode = 'w', $force = true)) {
+			return true;
+		}
+		$this->out('Unable to write spam protection file.');
+		CakeLog::error('CakephpQueue Monitor: Unable to write spam protection file.');
+		// if we can't write to the file, we can't send the alert because we could be spamming.
+		return false;
+	}
+
+	/**
+	 * Sends the message to HipChat
+	 *
+	 * @param string The message you want to send
+	 */
+	private function sendToHipchat($message) {
+		$hipchatToken = 'c7cdab9bff9a7e0aae766aea345334';
+		// $hipchatRoomID = '84232'; // Scrum: TribeHR
+		$hipchatRoomID = '2124722'; // QueueMinder Test
```

Comment:
Should this be set to the scrum room, or the test room?
Or should this be a bootstrap property?